QS: Add time-based upkeep (#1457)
smartcontractkit/documentation: src/content/chainlink-automation/tutorials/time-based-upkeep.mdx

1.  In Remix, within the Solidity Compiler tab, copy the ABI here:

    <ClickToZoom src="/images/automation/qs-time-based-upkeep/get-abi-remix.png" alt="Remix Copy ABI" />

1.  In the Chainlink Automation App, paste the ABI into the ABI field:

    <ClickToZoom
      src="/images/automation/qs-time-based-upkeep/couldnt-fetch-abi-ui.png"
      alt="Chainlink Automation App ABI field"
    />

</Fragment>
</TabsContent>

## Register a new Upkeep

<div class="remix-callout">
  <a href="https://automation.chain.link">Open the Chainlink Automation App</a>
</div>

1. **Click the Register New Upkeep button**

   <ClickToZoom src="/images/contract-devs/automation/auto-ui-home.png" alt="Click Register New Upkeep" />

1. **Connect your wallet** using the _Connect Wallet_ choose a network. For a list of supported networks, see the [Supported Blockchain Networks](/chainlink-automation/supported-networks) section. The Chainlink Automation App also lists the currently supported networks.

   <ClickToZoom src="/images/contract-devs/automation/auto-ui-wallet.png" alt="Connect With Metamask" />

1. Select the time-based trigger.

> 1. Provide the address of your deployed contract. If your contract is unverified, paste the ABI into the corresponding text box.

Reviewer: ```suggestion
1. Provide the address of your deployed contract. If your contract is not verified, paste the ABI into the corresponding text box.
```

1. If you're using the `Counter` contract, add a value for the function input in the Chainlink Automation App:

   <ClickToZoom src="/images/automation/qs-time-based-upkeep/update-amount.png" alt="Function input" />

>    Each time Automation triggers your function, based on the schedule you give it, it will run the `updateCounter` function, which increments the `counter` variable using the value you enter for `updateAmount`.

Reviewer: ```suggestion
   Each time Automation triggers your function, based on the schedule you give it, it will run the `updateCounter` function which increments the `counter` variable using the value you enter for `updateAmount`.
```

  to interact with Chainlink's systems and services so that you can integrate them into your own. This template is
  provided "AS IS" and "AS AVAILABLE" without warranties of any kind, has not been audited, and may be missing key
  checks or error handling to make the usage of the product more clear. Do not use the code in this example in a
  production environment without completing your own audits and application of best practices. Neither Chainlink Labs,
  the Chainlink Foundation, nor Chainlink node operators are responsible for unintended outputs that are generated due
  to errors in code.
</Aside>

## Objective

In this tutorial, you will learn how to set up a time-based upkeep to automate a smart contract function using Chainlink Automation.

You can use time-based upkeeps to:

- Schedule a rebalancing activity
- Schedule a recurring payment
- Schedule circuit breaker checks
- Automate triggering of any smart contract function that needs to run on a certain time schedule, or at a certain time in the future

## Before you begin

Before you start this tutorial, complete the following items:

> - Set up a cryptocurrency wallet such as [MetaMask](https://metamask.io/).
> - Fund your wallet with the following testnet tokens:
>   - The gas currency for the chain where your contract is deployed. For this example, get 0.5 Sepolia ETH from one of the available [public faucets](https://faucetlink.to/sepolia).
>   - ERC-677 LINK. [Get 20 testnet LINK](https://faucets.chain.link/).

Reviewer 2: ```suggestion
- If you are new to smart contract development, learn how to [Deploy Your First Smart Contract](/getting-started/deploy-your-first-contract).
- Set up a cryptocurrency wallet such as [MetaMask](https://metamask.io/).
- Fund your wallet with the following testnet tokens:
  - The gas currency for the chain where your contract is deployed. For this example, get 0.5 Sepolia ETH from one of the available [public faucets](https://faucetlink.to/sepolia).
  - ERC-677 LINK. [Get 20 testnet LINK](https://faucets.chain.link/).
```

### Entering upkeep details

To complete the upkeep registration process, you must enter some information about your upkeep including its name, gas limit, starting balance LINK, and contact information.

![Automation Upkeep Details](/images/contract-devs/automation/automation-upkeep-details.png)

1. **Complete the required details:**

   - **Upkeep name**: This will be publicly visible in the Chainlink Automation app.
   - **Gas limit**: This is the maximum amount of gas that your transaction requires to execute on chain. This value defaults to 500000, which is sufficient for the simple `updateCounter` function in the `Counter` contract. If the gas required to execute your transaction exceeds the gas limit that you specified, your transaction will not be confirmed.
   - **Starting balance (LINK)**: Specify a LINK starting balance to fund your upkeep. If you're using the `Counter` contract, 3 LINK is a sufficient starting balance. If your upkeep is underfunded after you create it, you can add more funds later.
   - **Your email address**: This email address will be encrypted and is used to send you an email when your upkeep is underfunded.

### Check your counter variable

> If you deployed the `Counter` contract, check the value of the counter variable in Sepolia Etherscan to see that it has been updated after 5 minutes. For this [example contract](https://sepolia.etherscan.io/address/0x1776d538436b476d9cbd7b30660bff3148831710#readContract) has the following value after being updated every 5 minutes for about one day:

Author: Updated to include both.
